### PR TITLE
Migrate broker SASL/SSL e2e tests to new rekt test framework

### PR DESCRIPTION
**Pull request by creydr**
Another PR of the series to migrate the e2e tests in `test/e2e_broker` to the new reconciler-test framework.

## Proposed Changes
- :broom: Migrate `TestBrokerAuthPlaintext` test
- :broom: Migrate `TestBrokerAuthSsl` test
- :broom: Migrate `TestBrokerAuthSaslPlaintextScram512` test
- :broom: Migrate `TestBrokerAuthSslSaslScram512` test

*Hints for reviewers:*
Tests for missing auth secret are added in #2968

### CODE REVIEW

File: test/rekt/features/broker_auth.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+	eventId := uuid.New().String()
+	eventToSend := test.MinEvent()
+	eventToSend.SetID(eventId)
+
+	f.Setup("Create auth secret", kafkaauthsecret.Install(authSecretName, authSecretOptions...))
+
+	f.Setup("Create broker config", brokerconfigmap.Install(brokerConfigName,
+		brokerconfigmap.WithNumPartitions(3),
+		brokerconfigmap.WithReplicationFactor(1),
```
Comment (pierDipi) [inline test/rekt/features/broker_auth.go:75]:
```suggestion
		brokerconfigmap.WithReplicationFactor(3),
```

This makes the test more reliable

**Review by pierDipi — COMMENTED** [review]
All the `TODO: find better way`, howerver, I'm not sure panic is that bad that is worth keeping the `TODOs`

File: test/rekt/features/broker_auth.go (hunk 1 above)
Comment (creydr) [reply test/rekt/features/broker_auth.go]: :+1: Done

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**